src/App/Controls/ExtendedDatePicker.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Bit.App.Controls
+{
+    public class ExtendedDatePicker : DatePicker
+    {
+        private string _format;
+        
+        public static readonly BindableProperty PlaceHolderProperty = BindableProperty.Create(
+            nameof(PlaceHolder), typeof(string), typeof(ExtendedDatePicker));
+        
+        public string PlaceHolder
+		{
+            get { return (string)GetValue(PlaceHolderProperty); }
+            set { SetValue(PlaceHolderProperty, value); }
+		}
+        
+        public static readonly BindableProperty NullableDateProperty = BindableProperty.Create(
+            nameof(NullableDate), typeof(DateTime?), typeof(ExtendedDatePicker));
+
+		public DateTime? NullableDate
+		{
+            get { return (DateTime?)GetValue(NullableDateProperty); }
+            set
+            {
+                SetValue(NullableDateProperty, value); 
+                UpdateDate();
+            }
+		}
+
+        private void UpdateDate()
+        {
+            if (NullableDate.HasValue)
+            {
+                if (_format != null)
+                {
+                    Format = _format;
+                }
+            }
+            else
+            {
+                Format = PlaceHolder;
+            }
+        }
+
+        protected override void OnBindingContextChanged()
+		{
+			base.OnBindingContextChanged();
+            if (BindingContext != null)
+			{
+				_format = Format;
+				UpdateDate();
+			}
+		}
+
+		protected override void OnPropertyChanged(string propertyName = null)
+		{
+			base.OnPropertyChanged(propertyName);
+
+			if (propertyName == DateProperty.PropertyName || (propertyName == IsFocusedProperty.PropertyName && !IsFocused && (Date.ToString("d") == DateTime.Now.ToString("d"))))
```

Comment:
Question: What's the significance of `DateTime.Now` in this context? Is that the default value of `DatePicker`? Would this impact if the user really wanted to choose `Today` but coupled with a different time component (so 1 hour from now)?

Reply:
This is a hack to get DatePicker to actually return the date when the user hits "done" without actually adjusting the date (i.e. same day).  The time is selected separately and the two are re-combined into a proper DateTime on the way out the door.